oak-core-spi/src/main/java/org/apache/jackrabbit/oak/spi/gc/DelegatingGCMonitor.java

```diff
@@ -41,15 +40,15 @@ public class DelegatingGCMonitor implements GCMonitor {
      * @param gcMonitors
      */
     public DelegatingGCMonitor(@NotNull Collection<? extends GCMonitor> gcMonitors) {
-        this.gcMonitors = newConcurrentHashSet();
+        this.gcMonitors = Collections.synchronizedSet(new HashSet<>());
```

Comment:
Shouldn't we use `ConcurrentHashMap.newKeySet()` ?

Reply:
Thanks.